pkg/sqlcache/informer/indexer.go

```diff
@@ -184,7 +185,13 @@ func (i *Indexer) Index(indexName string, obj any) ([]any, error) {
 	// HACK: sql.Statement.Query does not allow to pass slices in as of go 1.19 - create an ad-hoc statement
 	query := fmt.Sprintf(selectQueryFmt, db.Sanitize(i.GetName()), strings.Repeat(", ?", len(values)-1))
 	stmt := i.Prepare(query)
-	defer i.CloseStmt(stmt)
+
+	defer func() {
+		cerr := i.CloseStmt(stmt)
+		if cerr != nil {
+			err = errors.Join(err, cerr)
```

Comment:
Does the error message from `cerr` provide enough context to know that it came from this close statement call (or just allow us to troubleshoot)?

Reply:
Hopefully, but that really depends on the SQLite library implementation.

I could add the statement as context if you prefer that way.

Reply:
if it's not too much trouble yes that'd be great to have a bit more context, just in case.

Reply:
addressed as 7d290ce8

TIL: the current library actually never returns errors on close - but I still feel like this code is good to have, as it might change and/or we might switch to another library which might

current https://gitlab.com/cznic/sqlite/-/blob/v1.34.5/sqlite.go#L472-479
possible https://github.com/mattn/go-sqlite3/blob/7658c06970ecf5588d8cd930ed1f2de7223f1010/sqlite3.go#L1896-L1915